From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Igor Pecovnik <igor.pecovnik@****l.com>
Date: Wed, 24 May 2023 18:49:21 +0200
Subject: Patching u-boot rockchip64 files arch/arm/dts/rk3399-rock-pi-4.dtsi
 arch/arm/mach-rockchip/Kconfig board/rockchip/evb_rk3399/evb-rk3399.c
 configs/rock-pi-4-rk3399_defconfig drivers/ram/rockchip/sdram_common.c
 include/configs/rk3399_common.h include/debug_uart.h

Signed-off-by: Igor Pecovnik <igor.pecovnik@****l.com>
---
 arch/arm/dts/rk3399-rock-pi-4.dtsi     |  4 +-
 arch/arm/mach-rockchip/Kconfig         |  2 -
 board/rockchip/evb_rk3399/evb-rk3399.c |  2 +
 configs/rock-pi-4-rk3399_defconfig     | 21 ++++++----
 drivers/ram/rockchip/sdram_common.c    |  4 ++
 include/configs/rk3399_common.h        |  1 +
 include/debug_uart.h                   | 16 +++++++
 7 files changed, 39 insertions(+), 11 deletions(-)

diff --git a/arch/arm/dts/rk3399-rock-pi-4.dtsi b/arch/arm/dts/rk3399-rock-pi-4.dtsi
index b28888ea92..003498f12a 100644
--- a/arch/arm/dts/rk3399-rock-pi-4.dtsi
+++ b/arch/arm/dts/rk3399-rock-pi-4.dtsi
@@ -13,15 +13,15 @@
 / {
 	aliases {
 		mmc0 = &sdmmc;
 		mmc1 = &sdhci;
 	};
-
+/*
 	chosen {
 		stdout-path = "serial2:1500000n8";
 	};
-
+*/
 	clkin_gmac: external-gmac-clock {
 		compatible = "fixed-clock";
 		clock-frequency = <125000000>;
 		clock-output-names = "clkin_gmac";
 		#clock-cells = <0>;
diff --git a/arch/arm/mach-rockchip/Kconfig b/arch/arm/mach-rockchip/Kconfig
index 213d8d0dd6..0f2b063824 100644
--- a/arch/arm/mach-rockchip/Kconfig
+++ b/arch/arm/mach-rockchip/Kconfig
@@ -233,11 +233,10 @@ config ROCKCHIP_RK3399
 	select SPL_RAM if SPL
 	select SPL_REGMAP if SPL
 	select SPL_SYSCON if SPL
 	select TPL_NEEDS_SEPARATE_STACK if TPL
 	select SPL_SEPARATE_BSS
-	select SPL_SERIAL
 	select SPL_DRIVERS_MISC
 	select CLK
 	select FIT
 	select PINCTRL
 	select RAM
@@ -249,11 +248,10 @@ config ROCKCHIP_RK3399
 	imply PRE_CONSOLE_BUFFER
 	imply ROCKCHIP_COMMON_BOARD
 	imply ROCKCHIP_SDRAM_COMMON
 	imply SPL_ATF_NO_PLATFORM_PARAM if SPL_ATF
 	imply SPL_ROCKCHIP_COMMON_BOARD
-	imply TPL_SERIAL
 	imply TPL_LIBCOMMON_SUPPORT
 	imply TPL_LIBGENERIC_SUPPORT
 	imply TPL_SYS_MALLOC_SIMPLE
 	imply TPL_DRIVERS_MISC
 	imply TPL_OF_CONTROL
diff --git a/board/rockchip/evb_rk3399/evb-rk3399.c b/board/rockchip/evb_rk3399/evb-rk3399.c
index abb76585cf..a0eb5917e4 100644
--- a/board/rockchip/evb_rk3399/evb-rk3399.c
+++ b/board/rockchip/evb_rk3399/evb-rk3399.c
@@ -13,10 +13,12 @@
 #ifndef CONFIG_SPL_BUILD
 int board_early_init_f(void)
 {
 	struct udevice *regulator;
 	int ret;
+	
+	gd->flags |= (GD_FLG_SILENT | GD_FLG_DISABLE_CONSOLE);
 
 	ret = regulator_get_by_platname("vcc5v0_host", &regulator);
 	if (ret) {
 		debug("%s vcc5v0_host init fail! ret %d\n", __func__, ret);
 		goto out;
diff --git a/configs/rock-pi-4-rk3399_defconfig b/configs/rock-pi-4-rk3399_defconfig
index 1b3870cfc3..01486f8daf 100644
--- a/configs/rock-pi-4-rk3399_defconfig
+++ b/configs/rock-pi-4-rk3399_defconfig
@@ -8,16 +8,16 @@ CONFIG_ENV_SIZE=0x6000
 CONFIG_ENV_OFFSET=0x3F8000
 CONFIG_ENV_SECT_SIZE=0x1000
 CONFIG_DEFAULT_DEVICE_TREE="rk3399-rock-pi-4b"
 CONFIG_ROCKCHIP_RK3399=y
 CONFIG_TARGET_EVB_RK3399=y
-CONFIG_DEBUG_UART_BASE=0xFF1A0000
-CONFIG_DEBUG_UART_CLOCK=24000000
-CONFIG_SPL_SPI_FLASH_SUPPORT=y
-CONFIG_SPL_SPI=y
+# CONFIG_DEBUG_UART_BASE=0xFF1A0000
+# CONFIG_DEBUG_UART_CLOCK=24000000
+# CONFIG_SPL_SPI_FLASH_SUPPORT=n
+# CONFIG_SPL_SPI=n
 CONFIG_SYS_LOAD_ADDR=0x800800
-CONFIG_DEBUG_UART=y
+# CONFIG_DEBUG_UART is not set
 # CONFIG_ANDROID_BOOT_IMAGE is not set
 CONFIG_DEFAULT_FDT_FILE="rockchip/rk3399-rock-pi-4b.dtb"
 CONFIG_DISPLAY_BOARDINFO_LATE=y
 CONFIG_MISC_INIT_R=y
 # CONFIG_SPL_RAW_IMAGE_SUPPORT is not set
@@ -60,12 +60,12 @@ CONFIG_PMIC_RK8XX=y
 CONFIG_REGULATOR_PWM=y
 CONFIG_REGULATOR_RK8XX=y
 CONFIG_PWM_ROCKCHIP=y
 CONFIG_RAM_RK3399_LPDDR4=y
 CONFIG_DM_RESET=y
-CONFIG_BAUDRATE=1500000
-CONFIG_DEBUG_UART_SHIFT=2
+# CONFIG_BAUDRATE=1500000
+# CONFIG_DEBUG_UART_SHIFT=2
 CONFIG_ROCKCHIP_SPI=y
 CONFIG_SYSRESET=y
 CONFIG_USB=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_XHCI_DWC3=y
@@ -85,5 +85,12 @@ CONFIG_DM_VIDEO=y
 CONFIG_DISPLAY=y
 CONFIG_VIDEO_ROCKCHIP=y
 CONFIG_DISPLAY_ROCKCHIP_HDMI=y
 CONFIG_SPL_TINY_MEMSET=y
 CONFIG_ERRNO_STR=y
+CONFIG_BOOTDELAY=-2
+CONFIG_SILENT_CONSOLE=y
+CONFIG_SYS_DEVICE_NULLDEV=y
+CONFIG_SILENT_CONSOLE_UPDATE_ON_SET=y
+CONFIG_SILENT_CONSOLE_UPDATE_ON_RELOC=y
+CONFIG_SYS_CONSOLE_INFO_QUIET=y
+CONFIG_DISABLE_CONSOLE=y
diff --git a/drivers/ram/rockchip/sdram_common.c b/drivers/ram/rockchip/sdram_common.c
index b3e7421d08..bd943f383f 100644
--- a/drivers/ram/rockchip/sdram_common.c
+++ b/drivers/ram/rockchip/sdram_common.c
@@ -8,10 +8,14 @@
 #include <ram.h>
 #include <asm/io.h>
 #include <asm/arch-rockchip/sdram.h>
 #include <asm/arch-rockchip/sdram_common.h>
 
+#ifdef CONFIG_SPL_BUILD
+DEBUG_UART_FUNCS
+#endif
+
 #ifdef CONFIG_RAM_ROCKCHIP_DEBUG
 void sdram_print_dram_type(unsigned char dramtype)
 {
 	switch (dramtype) {
 	case DDR3:
diff --git a/include/configs/rk3399_common.h b/include/configs/rk3399_common.h
index 055e4cf592..25c3cd7684 100644
--- a/include/configs/rk3399_common.h
+++ b/include/configs/rk3399_common.h
@@ -61,10 +61,11 @@
 #endif
 
 #include <config_distro_bootcmd.h>
 #include <environment/distro/sf.h>
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"silent=1\0" \
 	ENV_MEM_LAYOUT_SETTINGS \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"partitions=" PARTS_DEFAULT \
 	ROCKCHIP_DEVICE_SETTINGS \
 	BOOTENV \
diff --git a/include/debug_uart.h b/include/debug_uart.h
index 714b369e6f..327056568a 100644
--- a/include/debug_uart.h
+++ b/include/debug_uart.h
@@ -129,10 +129,11 @@ void printdec(unsigned int value);
 		CONFIG_DEBUG_UART_SHIFT)
 
 /*
  * Now define some functions - this should be inserted into the serial driver
  */
+#ifdef CONFIG_DEBUG_UART
 #define DEBUG_UART_FUNCS \
 \
 	static inline void _printch(int ch) \
 	{ \
 		if (ch == '\n') \
@@ -195,6 +196,21 @@ void printdec(unsigned int value);
 		board_debug_uart_init(); \
 		_debug_uart_init(); \
 		_DEBUG_UART_ANNOUNCE \
 	} \
 
+#else
+// Add stub uart debug functions, for when the uart console is silenced
+#define DEBUG_UART_FUNCS \
+	void printch(int ch){} \
+	void printascii(const char *str){} \
+	void printhex1(unsigned int digit){} \
+	void printhex(unsigned int value, int digits){} \
+	void printhex2(unsigned int value){} \
+	void printhex4(unsigned int value){} \
+	void printhex8(unsigned int value){} \
+	void printdec(unsigned int value){} \
+	void debug_uart_init(void){} \
+	
+#endif
+
 #endif
-- 
Created with Armbian build tools https://github.com/armbian/build

